src/Parser.js

```diff
@@ -92,6 +92,12 @@ function assertFuncOrArg(parsed: ParsedFuncOrArgOrDollar): ParsedFuncOrArg {
     return parsed;
 }
 
+type GroupParser = (optional: boolean) => ?ParsedFuncOrArgOrDollar;
+const GROUP_PARSER_EXTENSIONS: { [type: string]: GroupParser } = {};
+export function defineGroupParser(type: string, parser: GroupParser) {
+    GROUP_PARSER_EXTENSIONS[type] = parser;
```

Comment:
Like this

Reply:
Interesting idea.  This could also be used for defining a custom parser for `\ce` from the `mhchem` package.